launch/launch/actions/append_environment_variable.py

```diff
@@ -72,18 +86,22 @@ def parse(
         cls,
         entity: Entity,
         parser: Parser,
-    ):
+    ) -> Tuple[Type[Self], AppendEnvironmentVariableParsedDict]:
         """Parse an 'append_env' entity."""
         _, kwargs = super().parse(entity, parser)
-        kwargs['name'] = parser.parse_substitution(entity.get_attr('name'))
-        kwargs['value'] = parser.parse_substitution(entity.get_attr('value'))
+
+        new_kwargs = AppendEnvironmentVariableParsedDict(
+            name=parser.parse_substitution(entity.get_attr('name')),
+            value=parser.parse_substitution(entity.get_attr('value')),
+            **kwargs)
```

Comment:
I think I'd prefer to keep it looking like it did before, i.e., something like:

```suggestion
        new_kwargs = AppendEnvironmentVariableParsedDict(**kwargs)
        new_kwargs['name'] = parser.parse_substitution(entity.get_attr('name'))
        new_kwargs['value'] = parser.parse_substitution(entity.get_attr('value'))
```

What do you think? This way all kwargs are set with `new_kwargs['_____']`

Same for all other actions

Reply:
Sadly doing it that way is considered untype safe since there are no guarentees that all the keys get added.
https://mypy-play.net/?mypy=latest&python=3.12&gist=67fac86e3320b4dc9002cc4344bd254a

Reply:
unlike `ActionParsedDict`, your example doesn't set `total=False`; setting it resolves the issue

Reply:
That is true for keys that are optional like in `ActionParsedDict`. But marking keys that always exists as optional is not ideal. It might be to remove the `TypedDict` as mentioned above.

Reply:
We can always start with a simple `Dict[str, Any]` and reconsider later.